Time parsing error

Error appearing after a while:
```
{"level":"info","time":"2020-06-24T06:39:29Z","message":"Starting HTTP server at ':8080'"}
{"level":"info","time":"2020-06-24T06:39:29Z","message":"Initializing HTTP server at ':8080'"}
{"level":"info","time":"2020-06-24T06:39:29Z","message":"Updating groups information"}
{"level":"info","time":"2020-06-24T06:39:29Z","message":"getting rules static content"}
{"level":"debug","time":"2020-06-24T06:39:29Z","message":"Connecting to http://ccx-insights-content-service:8080/api/v1/content "}
{"level":"debug","time":"2020-06-24T06:39:29Z","message":"Connecting to http://ccx-insights-content-service:8080/api/v1/groups "}
{"level":"info","time":"2020-06-24T06:39:29Z","message":"Received 4 groups"}
{"level":"info","time":"2020-06-24T06:39:29Z","message":"Updating groups configuration each 60.000000 seconds"}
{"level":"error","time":"2020-06-24T06:39:29Z","message":"invalid to parse time \"2020-02-07 14:19:00\" using layout \"2006-01-02T15:04:05Z07:00\""}
{"level":"info","time":"2020-06-24T06:39:37Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:39:43Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:39:47Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:39:53Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:39:57Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:40:03Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:40:07Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:40:13Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:40:17Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:40:23Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:40:27Z","message":"Request received - URI: /api/insights-results-aggregator/v1/openapi.json, Method: GET"}
{"level":"info","time":"2020-06-24T06:40:29Z","message":"getting rules static content"}
{"level":"debug","time":"2020-06-24T06:40:29Z","message":"Connecting to http://ccx-insights-content-service:8080/api/v1/content "}
{"level":"error","time":"2020-06-24T06:40:29Z","message":"invalid to parse time \"2019-10-29 15:00:00\" using layout \"2006-01-02T15:04:05Z07:00\""}
panic: close of closed channel

goroutine 12 [running]:
github.com/RedHatInsights/insights-results-smart-proxy/content.updateContent(0xc00002c033, 0x4b, 0xc00002a030, 0x30, 0xdf8475800)
	/opt/app-root/src/content/content.go:155 +0x103
github.com/RedHatInsights/insights-results-smart-proxy/content.RunUpdateContentLoop(0xc00002c033, 0x4b, 0xc00002a030, 0x30, 0xdf8475800)
	/opt/app-root/src/content/content.go:139 +0x57
created by main.startServer
	/opt/app-root/src/smart_proxy.go:101 +0x240
```

--- a/content/content_test.go
+++ b/content/content_test.go
@@ -1,0 +1,45 @@
+// Copyright 2020 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package content provides API to get rule's content by its `rule id` and `error key`.
+// It takes all the work of caching rules taken from content service
+package content_test
+
+import (
+	"testing"
+
+	cs_content "github.com/RedHatInsights/insights-content-service/content"
+	"github.com/RedHatInsights/insights-operator-utils/tests/helpers"
+	"github.com/RedHatInsights/insights-results-aggregator-data/testdata"
+
+	"github.com/RedHatInsights/insights-results-smart-proxy/content"
+)
+
+func TestUpdateContentBadTime(t *testing.T) {
+	// using testdata.RuleContent4 because contains datetime in a different format
+	ruleContentDirectory := cs_content.RuleContentDirectory{
+		Config: cs_content.GlobalRuleConfig{
+			Impact: testdata.ImpactStrToInt,
+		},
+		Rules: map[string]cs_content.RuleContent{
+			"rc4": testdata.RuleContent4,
+		},
+	}
+
+	content.LoadRuleContent(&ruleContentDirectory)
+	close(content.RuleContentDirectoryReady)
+	_, err := content.GetRuleWithErrorKeyContent(
+		testdata.Rule4ID, testdata.ErrorKey4)
+	helpers.FailOnError(t, err)
+}

--- a/content/export_test.go
+++ b/content/export_test.go
@@ -1,0 +1,20 @@
+// Copyright 2020 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package content
+
+var (
+	LoadRuleContent           = loadRuleContent
+	RuleContentDirectoryReady = ruleContentDirectoryReady
+)

--- a/content/parsing.go
+++ b/content/parsing.go
@@ -23,6 +23,10 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+const (
+	timeParseFormat = "2006-01-02 15:04:05"
+)
+
 // TODO: consider moving parsing to content service
 
 // loadRuleContent loads the parsed rule content into the storage
@@ -49,13 +53,9 @@ func loadRuleContent(contentDir *content.RuleContentDirectory) {
 				return
 			}
 
-			publishDate, err := time.Parse(time.RFC3339, errorProperties.Metadata.PublishDate)
+			publishDate, err := timeParse(errorProperties.Metadata.PublishDate)
+
 			if err != nil {
-				log.Error().Msgf(
-					`invalid to parse time "%v" using layout "%v"`,
-					errorProperties.Metadata.PublishDate,
-					time.RFC3339,
-				)
 				return
 			}
 
@@ -99,4 +99,22 @@ func commaSeparatedStrToTags(str string) []string {
 	}
 
 	return strings.Split(str, ",")
+}
+
+func timeParse(value string) (time.Time, error) {
+	var err error
+	for _, datetimeLayout := range []string{timeParseFormat, time.RFC3339} {
+		parsedDate, err := time.Parse(datetimeLayout, value)
+
+		if err == nil {
+			return parsedDate, nil
+		}
+
+		log.Error().Msgf(
+			`invalid to parse time "%v" using layout "%v"`,
+			value, datetimeLayout,
+		)
+	}
+
+	return time.Time{}, err
 }

--- a/go.mod
+++ b/go.mod
@@ -7,7 +7,7 @@ require (
 	github.com/RedHatInsights/insights-content-service v0.0.0-20200619153839-23a428468a08
 	github.com/RedHatInsights/insights-operator-utils v1.2.0
 	github.com/RedHatInsights/insights-results-aggregator v0.0.0-20200622123018-31c3f8ac30bd
-	github.com/RedHatInsights/insights-results-aggregator-data v0.0.0-20200622112651-e5a7c1d79015
+	github.com/RedHatInsights/insights-results-aggregator-data v0.0.0-20200624135606-84f78db90ac2
 	github.com/RedHatInsights/insights-results-aggregator-utils v0.0.0-20200616074815-67f30b0e724d // indirect
 	github.com/Shopify/sarama v1.26.4 // indirect
 	github.com/aws/aws-sdk-go v1.32.6 // indirect

